src/internal/deploy/assets/postgres.go

```diff
@@ -320,6 +383,19 @@ func PostgresStatefulSet(opts *AssetOpts, backend Backend, diskSpace int) interf
 	}
 }
 
+// TestPostgresVolume creates a persistent volume for use with StatefulSets in a
```

Comment:
Is there a way to create a statefulset with the volumes pre-defined? My understanding was that we would need a local storage class to do the local testing with a statefulset. Something I noticed when looking at this last was that we really shouldn't be using deployments for etcd or postgres because we run the risk of deadlocks due to the `ReadWriteOnce` access mode for the volume. It would be good if we could move all of this over to statefulsets and it looks like the local storage class may give us a way to move the local deployment to statefulsets as well.

Reply:
I agree that we should move these all over to statefulsets, and I'd be happy to do that once I have some spare cycles, but it's a bit out of scope for me now.  I'd like it if we only did statefulsets for pachd as well, just to reduce the surface area of the assets code.

Reply:
Yeah, definitely out of scope for this PR. I just wasn't sure if this function would get any use since I'm not aware of a way to set up a statefulset with pre-defined volumes. Also, we should probably keep pachd as a deployment since it is not stateful.